src/software/thunderscope/robot_communication.py

```diff
@@ -109,41 +141,48 @@ def run(self):
                     ),
                 )
 
-                primitive_set = PrimitiveSet(
-                    time_sent=Timestamp(epoch_timestamp_seconds=time.time()),
-                    stay_away_from_ball=False,
-                    robot_primitives={
-                        robot_id: Primitive(direct_control=diagnostics_primitive)
-                        for robot_id in self.robots_connected_to_diagnostics
-                    },
-                    sequence_number=self.sequence_number,
-                )
+                # for all robots connected to diagnostics, set their primitive
+                for robot_id in self.robots_connected_to_diagnostics:
+                    robot_primitives[robot_id] = Primitive(
+                        direct_control=diagnostics_primitive
+                    )
+
+            # initialize total primitive set and send it
+            primitive_set = PrimitiveSet(
+                time_sent=Timestamp(epoch_timestamp_seconds=time.time()),
+                stay_away_from_ball=False,
+                robot_primitives=robot_primitives,
+                sequence_number=self.sequence_number,
+            )
 
-                self.sequence_number += 1
+            self.sequence_number += 1
 
-                if self.estop_reader.isEstopPlay():
-                    self.last_time = primitive_set.time_sent.epoch_timestamp_seconds
-                    self.send_primitive_set.send_proto(primitive_set)
+            if self.estop_reader.isEstopPlay():
+                self.last_time = primitive_set.time_sent.epoch_timestamp_seconds
+                self.send_primitive_set.send_proto(primitive_set)
 
-                time.sleep(0.001)
+            time.sleep(0.001)
```

Comment:
Was just playing around and noticed that we're broadcasting over 1000 primitivesets a second. We should definitely reduce this to avoid network congestion and also overwhelming Thunderloop which random primitives that will just be overwritten by the next one.
```suggestion
            time.sleep(0.01)
```